src/plugins/intel_cpu/src/transformations/snippets/x64/op/brgemm_cpu.cpp

```diff
@@ -180,5 +181,85 @@ size_t BrgemmCPU::get_offset_scratch() const {
     return get_input_offset(2);
 }
 
+BrgemmCPU::ShapeInfer::ShapeInfer(const std::shared_ptr<ov::Node>& n) {
+    const auto& brg = ov::as_type_ptr<BrgemmCPU>(n);
+    OPENVINO_ASSERT(brg, "Got invalid node in BrgemmCPU::ShapeInfer");
+
+    const auto brgemm_copy = brg->is_with_data_repacking() ? brg->get_brgemm_copy() : nullptr;
+    auto register_in_layout =  [this](const ov::Input<ov::Node>& in) {
+        m_layouts.emplace_back(snippets::lowered::PortDescriptorUtils::get_port_descriptor_ptr(in)->get_layout());
+    };
+
+    register_in_layout(n->input(0));
+    if (brgemm_copy)
+        register_in_layout(brgemm_copy->input(0));
+    else
+        register_in_layout(n->input(1));
+    m_layouts.emplace_back(snippets::lowered::PortDescriptorUtils::get_port_descriptor_ptr(n->output(0))->get_layout());
+}
+
+snippets::IShapeInferSnippets::Result
+BrgemmCPU::ShapeInfer::infer(const std::vector<std::reference_wrapper<const IShapeInferSnippets::VectorDims>>& input_shapes) {
+    OPENVINO_ASSERT(input_shapes.size() == 2, "BRGEMM expects 2 input shapes for shape inference");
```

Comment:
What confuses me is that the shape inference code is repeated almost one in one 2 times in `Brgemm` and `BrgemmCPU` nodes (Actually we have and 3rd place - `Brgemm::get_planar_output_shape` - but this is a function with `PartialShape` as a parameter so it is more difficult to extract common parts). 
Can we try to extract common computations in a separate method and reuse it? I suggest discussing this offline

Reply:
Valid point, thank you!
I simply derived `BrgemmCPU::SnapeInfer` from `Brgemm::SnapeInfer`